File: contains_duplicate/contains_duplicate.js
function contains_duplicate(nums) {
  const set = new Set();

  for (let i = 0; i < nums.length; i++) {
    if (set.has(nums[i])) return true;
    else set.add(nums[i]);
  }

  return false;
}

console.log(contains_duplicate([1, 2, 3, 1]));

File: contains_duplicate/contains_duplicate.py
def contains_duplicate(nums):
    s = set()

    for num in nums:
        if (num in s):
            return True
        else:
            s.add(num)

    return False


print(contains_duplicate([1, 2, 3, 4, 5, 6, 7, 8, 9, 1]))
